--- main.cpp	2014-10-29 10:08:10.000000000 -0400
+++ main.cpp	2015-05-11 01:32:33.995061821 -0400
@@ -197,14 +197,14 @@
 			(odtone::sap::kConf_File, po::value<std::string>()->default_value("link_sap.conf"), "Configuration file")
 			(odtone::sap::kConf_Receive_Buffer_Len, po::value<uint>()->default_value(4096), "Receive buffer length")
 			(odtone::sap::kConf_MIH_SAP_id, po::value<std::string>()->default_value("link"), "Link SAP ID")
-			(odtone::sap::kConf_Port, po::value<ushort>()->default_value(1235), "Listening port")
+			(odtone::sap::kConf_Port, po::value<odtone::ushort>()->default_value(1235), "Listening port")
 			(odtone::sap::kConf_Tec, po::value<std::string>()->default_value(""), "Technology type")
 			(odtone::sap::kConf_Interface_Addr, po::value<std::string>()->default_value(""), "Interface address")
 			(odtone::sap::kConf_Evt_List, po::value<std::string>()->default_value(""), "List of supported events")
 			(odtone::sap::kConf_Cmd_List, po::value<std::string>()->default_value(""), "List of supported commands")
 			(odtone::sap::kConf_MIHF_Id, po::value<std::string>()->default_value("local-mihf"), "Local MIHF ID")		
 			(odtone::sap::kConf_MIHF_Ip, po::value<std::string>()->default_value("127.0.0.1"), "Local MIHF IP")						
-			(odtone::sap::kConf_MIHF_Local_Port, po::value<ushort>()->default_value(1025), "Local MIHF communication port")
+			(odtone::sap::kConf_MIHF_Local_Port, po::value<odtone::ushort>()->default_value(1025), "Local MIHF communication port")
 		;
 
 		odtone::mih::config cfg(desc);
